keps/sig-api-machinery/4020-unknown-version-interoperability-proxy/README.md

```diff
@@ -276,76 +280,54 @@ This might be a good place to talk about core concepts and how they relate.
+    group-version(LocalDiscovery cache). This will be done via a discovery call
+    using a loopback client. A post-start hook will populate this cache, guaranteeing
+    the apiserver has a complete view of its served resources before processing
+    any incoming requests.
+    - an informer cache of resources served by each peer apiserver in the
+    cluster(PeerAggregatedDiscovery cache). This cache will be updated by an informer
+    on [apiserver identity Lease objects](https://github.com/kubernetes/enhancements/blob/master/keps/sig-api-machinery/1965-kube-apiserver-identity/README.md#proposal).
+    The informer's event handler will make discovery calls to peer apiservers
+    whose lease objects are created or updated.
 
 2. This filter will pass on the request to the next handler in the local aggregator
 chain, if:
    1. It is a non resource request
-   2. The StorageVersion informer cache hasn't synced yet or if `StorageVersionManager.Completed()`
-   has returned false. We
-   will serve error 503 in this case
-   3. The request has a header `X-Kubernetes-APIServer-Rerouted:true` that
-   indicates that this request has been proxied once
-   already. If for some reason the resource is not found locally, we will serve
-   error 503
-   4. No StorageVersion was retrieved for it, meaning the request is for an
-   aggregated API or for a custom resource
-   5. If the local apiserver ID is found in the list of serviceable-by server IDs
-   from the internal map
-
-3. If the local apiserver ID is not found in the list of serviceable-by server
-IDs, a random apiserver ID will be selected
-from the retrieved list and the request will be proxied to this apiserver
-
-4. If there is no apiserver ID retrieved for the requested GVR, we will serve
-404 with error `GVR <group_version_resource> is
-not served by anything in this cluster`
+   2. The LocalDiscovery cache or the apiserver identity lease informer hasn't synced
+   yet. We will serve error 503 in this case
+   3. The request has a header `X-Kubernetes-APIServer-Rerouted:true` that indicates
+   that this request has been proxied once already. If for some reason the
+   resource is not found locally, we will serve error 503
+   4. The requested resource was listed in the LocalDiscovery cache
+   5. No other peer apiservers were found to exist in the cluster
+
+3. If the requested resource was not found in the LocalDiscovery cache, we will
+try to fetch the resource from the PeerAggregatedDiscovery cache. The request
+will then be proxied to any peer apiserver, selected randomly, thats found to be
+able to serve the resource as indicated in the PeerAggregatedDiscovery cache.
+
+4. If there is no eligible apiserver found in the PeerAggregatedDiscovery cache
+for the requested resource, we will pass on the request to the next handler in the
+handler chain. This will either
+
+    - be eventually handled by the apiextensions-apiserver
```

Comment:
Can you add a section or example on how CRDs/aggregated resources are handled if they are added after the local cache is established?

Reply:
Done as point 3.1 in this section